### PR TITLE
support jdq avro 

### PR DESCRIPTION
**Pull request by gejibin**
to support jdq avro

### CODE REVIEW

**Comment by liuzix** [comment]
Could you update the pull request description according to the template? https://github.com/pingcap/ticdc/blob/master/.github/pull_request_template.md
Also please fix the linting error. Thanks!

File: cdc/sink/codec/jdq_avro.go [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+		if err != nil {
+			log.Warn("AppendRowChangedEvent: could not construct Jdq envelope", zap.String("table", e.Table.String()))
+			return EncoderNoOperation, errors.Annotate(err, "AppendRowChangedEvent could not construct Jdq envelope")
+		}
+
+		mqMessage.Value = evlp
+	} else {
+		// log.Warn("AppendRowChangedEvent: don't deal with it", zap.String("table", e.Table.String()))
+		// return EncoderNoOperation, nil
+		panic("AppendRowChangedEvent: row events error!!!")
```
Comment (liuzix) [inline cdc/sink/codec/jdq_avro.go:111]: Could you change to log.Panic and print more details?

File: cdc/sink/codec/jdq_avro.go [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+	}
+
+	// message key: primary keys, separated by space
+	keyStr := ""
+	pkeyCols := e.HandlePrimaryKeyColumns()
+	for _, key := range pkeyCols {
+		data, err := a.columnToJdqNativeData(key)
+		if err != nil {
+			log.Warn("AppendRowChangedEvent: avro encoding primary key failed", zap.String("table", e.Table.String()))
```
Comment (liuzix) [inline cdc/sink/codec/jdq_avro.go:120]: Could you change the log? Since we are not actually encoding the keys.

File: cdc/sink/codec/jdq_avro.go [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+                              {"name":"tab","type":"string"},
+                              {"name":"opt","type":"string"},
+                              {"name":"ts","type":"long"},
+                              {"name":"err","type":["string","null"]},
+                              {"name":"src","type":[{"type":"map","values":["string","null"]},"null"]},
+                              {"name":"cur","type":[{"type":"map","values":["string","null"]},"null"]},
+                              {"name":"cus","type":[{"type":"map","values":["string","null"]},"null"]}]}
+                  `
+
+	codec, err := goavro.NewCodec(jdwSchema)
```
Comment (liuzix) [inline cdc/sink/codec/jdq_avro.go:217]: It's better to cache the codec.

**Comment by hi-rustin** [comment]
@gejibin The PR is in conflict, can you help resolve the conflict and we can move forward with the PR? If you don't have time to continue working on it, please also close the PR. Thank you for your contribution! 💚 💙 💜 💛 ❤️

In addition, you will need to sign a cla agreement. You can refer to the robot's response to sign it. https://github.com/pingcap/ticdc/pull/1449#issuecomment-784942202

**Comment by hi-rustin** [comment]
/close 

I am closing this PR because it has not been updated for a long time.
Please feel free to contact me to reopen the PR if you have time to finish it. 
Thank you very much for your contribution!

**Pull request closed without merging**